crates/fuel-core/src/service/adapters/consensus_parameters_provider.rs

```diff
@@ -38,6 +41,7 @@ pub struct SharedState {
         SharedMutex<ConsensusParametersVersion>,
     pub(crate) consensus_parameters:
         SharedMutex<HashMap<ConsensusParametersVersion, Arc<ConsensusParameters>>>,
+    pub(crate) latest_stf_version: SharedMutex<StateTransitionBytecodeVersion>,
```

Comment:
May I also ask you to replace `Mutex`  with `RwLock` for `consensus_parameters` and `latest_stf_version` please?=)

Since we only read in most of the cases and write happens once per month

Reply:
Done in https://github.com/FuelLabs/fuel-core/commit/d8ea573b14fe0092ebd4c6dc75556e223313973e